### PR TITLE
Fix clone_model

### PR DESCRIPTION
**Pull request by Mistariano**
### Summary

Try to fix a bug inside `keras.models._clone_functional_model`, which can cause an error when clone a model containing a layer with 2 or more outputs

I'm not sure whether or not the issue needs a new unit test.. In fact I faced it when I'm using `multi_gpu_model` as mentioned in my earlier issue，so maybe it's a better idea to do some more test on 
`multi_gpu_model` instead (but I have no idea about that...)

### Related Issues

https://github.com/keras-team/keras/issues/12202

### PR Overview

- [y] This PR requires new unit tests [y/n] (make sure tests are included)
- [n] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y] This PR is backwards compatible [y/n]
- [n] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: keras/models.py [hunk 1]
```diff
@@ -139,7 +139,8 @@ def _clone_functional_model(model, input_tensors=None):
                         layer(computed_tensor, **kwargs))
                     output_masks = to_list(
                         layer.compute_mask(computed_tensor,
-                                           computed_mask))
+                                           computed_mask)) \
+                        if layer.supports_masking else [None] * len(output_tensors)
```
Comment (fchollet) [inline keras/models.py]: This should be structured as an `if` block for better readability:

```
if ...:
   output_masks = ...
else:
   output_masks = ...
```

You can simply specify a lack of mask by passing `None` (it doesn't have to be a list of None)

Comment (Mistariano) [reply keras/models.py]: Thanks for pointing that out! I agree with you that using `if` blocks is better and I'll commit it soon. 
But I think to assign a list of `None` is necessary, because `output_masks` will be passed to `zip` as the second argument later, so in my opinion it should be an iterable object but not a `None` (otherwise it can cause an error).

**Comment by Mistariano** [comment]
I added a test case but I'm not sure whether it's ok or not.
Have a look please! :)

**Comment by pfk-beta** [comment]
Is this PR ready to merge?
